sdk/src/main/scala/com/horizen/forge/Forger.scala

```diff
@@ -1,162 +1,166 @@
-            case Success(block) =>
-              sidechainNodeViewHolderRef ! LocallyGeneratedModifier[SidechainBlock](block)
-              sender() ! Success(block.id)
-            case Failure(e) =>
-              sender() ! Failure(e)
-          }
-        case Failure(e) =>
-          sender() ! Failure(new Exception(s"Unable to collect information for block template creation: ${e.getMessage}"))
-      }
+  protected def processStartForgingMessage: Receive = {
+    case StartForging => {
+      log.info("Receive StartForging message")
+      forgingIsActive = true
+      tryToCreateBlockNow()
+      sender() ! Success()
+    }
   }
 
-  override def receive: Receive = {
-    tryGetBlockTemplate orElse trySubmitBlock orElse tryForgeNextBlock orElse {
-      case message: Any => log.error("Forger received strange message: " + message)
+  protected def processStopForgingMessage: Receive = {
+    case StopForging => {
+      log.info("Receive StopForging message")
+      forgingIsActive = false
+      sender() ! Success()
     }
   }
-}
 
-object Forger extends ScorexLogging {
-  import scorex.core.NodeViewHolder.ReceivableMessages.GetDataFromCurrentView
+  protected def processTryForgeNextBlockForEpochAndSlotMessage: Receive = {
+    case TryForgeNextBlockForEpochAndSlot(epochNumber, slotNumber) => tryToCreateBlockForEpochAndSlot(epochNumber, slotNumber, Some(sender()))
+  }
 
-  object ReceivableMessages {
+  protected def tryToCreateBlockNow(): Unit = {
+    val currentTime: Long = Instant.now.getEpochSecond
+    val epochAndSlot = timestampToEpochAndSlot(currentTime)
+    log.info(s"Send TryForgeNextBlockForEpochAndSlot message with epoch and slot ${epochAndSlot}")
+    tryToCreateBlockForEpochAndSlot(epochAndSlot.epochNumber, epochAndSlot.slotNumber, None)
+  }
 
-    case object TryGetBlockTemplate
-    case object TryForgeNextBlock
-    case class TrySubmitBlock(blockBytes: Array[Byte])
+  protected def tryToCreateBlockForEpochAndSlot(epochNumber: ConsensusEpochNumber, slot: ConsensusSlotNumber, respondsToOpt: Option[ActorRef]): Unit = {
+    val forgeMessage: ForgeMessageBuilder#ForgeMessageType = forgeMessageBuilder.buildForgeMessageForEpochAndSlot(epochNumber, slot)
+    val forgedBlockAsFuture = (viewHolderRef ? forgeMessage).asInstanceOf[Future[ForgeResult]]
+    forgedBlockAsFuture.onComplete{
+      case Success(ForgeSuccess(block)) => {
+        log.info(s"Got successfully forged block with id ${block.id}")
+        viewHolderRef ! LocallyGeneratedModifier[SidechainBlock](block)
+        respondsToOpt.map(respondsTo => respondsTo ! Success(block.id))
+      }
+
+      case Success(SkipSlot) => {
+        log.info(s"Slot is skipped")
+        respondsToOpt.map(respondsTo => respondsTo ! Failure(new RuntimeException("Slot had been skipped")))
+      }
+
+      case Success(ForgeFailed(ex)) => {
+        log.info("Forging had been failed")
```

Comment:
this should be logged as an error and moreover should be managed in a way to let the node "owner" to detect the failure also in automatic mode.